diff --git a/virttest/libvirt_xml/vm_xml.py b/virttest/libvirt_xml/vm_xml.py
index 1a77bc2..1c98042 100644
--- a/virttest/libvirt_xml/vm_xml.py
+++ b/virttest/libvirt_xml/vm_xml.py
@@ -528,6 +528,8 @@ class VMXML(VMXMLBase):
         vmxml = VMXML(virsh_instance=virsh_instance)
         vmxml['xml'] = virsh_instance.dumpxml(vm_name,
                                               extra=options).stdout.strip()
+        vmxml.xmltreefile.remove_by_xpath('/os/nvram', remove_all=True)
+        vmxml.xmltreefile.write()
         return vmxml
 
     @staticmethod
@@ -551,7 +553,7 @@ class VMXML(VMXMLBase):
         """
         return librarian.get(type_name)
 
-    def undefine(self, options=None):
+    def undefine(self, options="--nvram"):
         """Undefine this VM with libvirt retaining XML in instance"""
         return self.virsh.remove_domain(self.vm_name, options)
 
diff --git a/virttest/virsh.py b/virttest/virsh.py
index 70dc302..5cb5e83 100644
--- a/virttest/virsh.py
+++ b/virttest/virsh.py
@@ -1302,7 +1302,7 @@ def define(xml_path, **dargs):
     return command(cmd, **dargs)
 
 
-def undefine(name, options=None, **dargs):
+def undefine(name, options="--nvram", **dargs):
     """
     Return cmd result of domain undefine (after shutdown/destroy).
 
